### DISCUSSION TITLE
difference between render and action without parameters

### DISCUSSION THREAD

**Discussion by orumetsu** [replit/kaboom#191] [Q&A] [answered]
I still don't understand what the difference is between action(( )=>{ }) and render(( )=>{ }), is there any difference? (I know it's mentioned in the website that it has something to do with update( ) but I still don't understand) and I'd alsolike to know what happens if there are tags and parameters on those functions, thanks!

**Comment by slmjkdbtl** [comment, marked as answer]
There's no difference, except all `action` are called before `render` (a lot of games want to run state updates and rendering separately)

**Reply by orumetsu** [reply]
ah i see, thank you!
